build_runner/lib/src/package_graph/dependency_ordering.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:math' show min;
+
+/// Finds the strongly connected components of an ordered graph using Tarjan's
+/// algorithm.
+///
+/// [V] is the type of values in the graph vertices. [K] must be a type suitable
+/// for using as a Map or Set key, and [key] must provide a consistent key for
+/// ever vertex. [successors] should return the children for a given vertex.
+///
+/// [nodes] must contain at least a root of every tree in the graph if there are
+/// disjoint subgraphs but it may contain all nodes.
+List<List<V>> stronglyConnectedComponents<K, V>(
+    Iterable<V> nodes, K Function(V) key, Iterable<V> Function(V) successors) {
```

Comment:
nit: `children` instead of `successors`? that is the more common term afaik.

Reply:
Done